client-spark/common/src/main/java/org/apache/spark/shuffle/reader/RssShuffleDataIterator.java

```diff
@@ -109,24 +123,20 @@ public boolean hasNext() {
       shuffleReadMetrics.incFetchWaitTime(fetchDuration);
       if (compressedData != null) {
         shuffleReadMetrics.incRemoteBytesRead(compressedData.limit() - compressedData.position());
-        // Directbytebuffers are not collected in time will cause executor easy 
-        // be killed by cluster managers(such as YARN) for using too much offheap memory
-        if (uncompressedData != null && uncompressedData.isDirect()) {
-          try {
-            RssShuffleUtils.destroyDirectByteBuffer(uncompressedData);
-          } catch (Exception e) {
-            throw new RssException("Destroy DirectByteBuffer failed!", e);
-          }
+
+        int uncompressedLen = compressedBlock.getUncompressLength();
+        if (uncompressedData == null || uncompressedData.capacity() < uncompressedLen) {
+          uncompressedData = ByteBuffer.allocate(uncompressedLen);
```

Comment:
In original implementation, the bytebuffer will be destoryed and recreate. So to avoid the frequent GC, it use the offheap-bytebuffer. 

And in this PR, we will recycle the bytebuffer, so I think it's no need to use the off-heap memory now. Maybe we should add the off-heap support in the next PR.

Reply:
PTAL @jerqi . This is the different with the original implementation.

Reply:
It's ok for me.